api/src/wfl/service/postgres.clj

```diff
@@ -71,6 +72,18 @@
       (remove (comp finished? :status))
       (run! #(update! % (maybe-cromwell-status %))))))
 
+(defn update-terra-workflow-statuses!
+  "Use `tx` to update `status` of `workflows` in `_workload`."
+  [tx {:keys [cromwell items workflows project] :as _workload}]
+  (letfn [(update! [{:keys [id uuid]} status]
+              (jdbc/update! tx items
+                            {:updated (OffsetDateTime/now) :uuid uuid :status status}
+                            ["id = ?" id]))]
+      (->> workflows
+           (remove (comp nil? :uuid))
+           (remove (comp finished? :status))
+           (run! #(update! % (terra/get-workflow-status-by-entity cromwell project %))))))
+
```

Comment:
This code is really similar to the one for cromwell - I wonder if we can re-use some of the logic by creating a higher-order function:

```clojure
(defn ^:private make-update-workflows [get-status!]
  (fn [tx {:keys [items workflows] :as workload}]
    (letfn [(update! [{:keys [id uuid]} status]
              (jdbc/update! tx items
                  {:updated (OffsetDateTime/now) :uuid uuid :status status}
                  ["id = ?" id]))]
       (->> workflows
         (remove (comp nil? :uuid))
         (remove (comp finished? :status))
         (run! #(update! % (get-status! workload %)))))))

(def update-workflow-statuses!
  (letfn [(get-cromwell-status [{:keys [cromwell]} {:keys [uuid]}]
            (if (util/uuid-nil? uuid) "skipped"
                                      (cromwell-status cromwell uuid)))]
    (make-update-workflows get-cromwell-status)))

(def update-terra-workflow-statuses!
  (letfn [(get-terra-status [{:keys [cromwell project]} workflow]
            (terra/get-workflow-status-by-entity cromwell project workflow))]
    (make-update-workflows get-terra-status)))
```